modules/entity_version_history/src/Form/HistoryTabSettingsForm.php

```diff
@@ -0,0 +1,209 @@
+          ],
+        ],
+        '#default_value' => is_array($history_configs[$entity_type_id]) ? array_keys($history_configs[$entity_type_id]) : [],
+      ];
+
+      // Create select list of the version fields in the bundle.
+      foreach ($bundles as $bundle_name => $label) {
+        $form['settings']["$entity_type_id-$bundle_name"] = [
+          '#title' => $label,
+          '#type' => 'select',
+          '#options' => $field_labels[$entity_type_id][$bundle_name],
+          '#states' => [
+            'visible' => [
+              ':input[name="entity_types[' . $entity_type_id . ']"]' => ['checked' => TRUE],
+              ':input[name="' . $entity_type_id . '[' . $bundle_name . ']"]' => ['checked' => TRUE],
+            ],
+          ],
+          '#default_value' => is_array($history_configs[$entity_type_id][$bundle_name]) ? array_keys($history_configs[$entity_type_id][$bundle_name]) : [],
+        ];
+      }
+    }
+
+    $form['actions']['#type'] = 'actions';
+    $form['actions']['submit'] = [
+      '#type' => 'submit',
+      '#value' => $this->t('Save configuration'),
+      '#button_type' => 'primary',
+    ];
+
+    return $form;
+  }
+
+  /**
+   * {@inheritdoc}
+   *
+   * @SuppressWarnings(PHPMD.CyclomaticComplexity)
+   */
+  public function submitForm(array &$form, FormStateInterface $form_state) {
+    foreach ($form_state->getValue('entity_types') as $target_entity_type_id => $entity_id_value) {
+      if (!$entity_id_value) {
+        // Delete all existing config settings with this entity id if the
+        // entity type (top level) checkbox is unchecked in the form.
+        if ($configs = HistoryTabSettings::loadByEntityType($target_entity_type_id)) {
+          $this->entityTypeManager->getStorage('entity_version_history_settings')->delete($configs);
+          continue;
+        }
+      }
+
+      foreach ($form_state->getValue($target_entity_type_id) as $target_bundle => $bundle_value) {
+        $config = HistoryTabSettings::loadByEntityTypeBundle($target_entity_type_id, $target_bundle);
+        if (!$bundle_value && $config) {
+          // Delete existing config setting with this entity id and bundle
+          // if the bundle checkbox is unchecked in the form.
+          $config->delete();
+          continue;
+        }
+
+        if ($config) {
+          // If the config exist already, skip creating the same config.
+          continue;
+        }
+
+        $target_field = $form_state->getValue("$target_entity_type_id-$target_bundle");
+        if ($target_field && $bundle_value) {
+          // If we have a target field and a bundle is checked, we create
+          // the new config entity.
+          $config = HistoryTabSettings::create([
+            'target_entity_type_id' => $target_entity_type_id,
+            'target_bundle' => $target_bundle,
+            'target_field' => $target_field,
+          ]);
+          $config->save();
```

Comment:
I see absolutely no confirmation message that something happened when I submit the form.